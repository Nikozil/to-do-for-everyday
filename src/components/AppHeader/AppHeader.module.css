.header {
  grid-area: h;
  background-color: #fff;
  display: flex;
  justify-content: space-between;
}
.username {
  margin-left: 5px;
}
.login {
  margin-right: 10px;
}
