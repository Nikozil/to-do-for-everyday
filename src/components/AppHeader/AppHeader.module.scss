@import '../../App.scss';

.header {
  grid-area: h;
  background-color: $header-color;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.pagename {
  margin-left: 5px;
  font-size: 1.5rem;
  width: 300px;

  @media screen and (max-width: 767px) {
    font-size: 1.2rem;
    width: 150px;
  }
}

.usermenu {
  display: flex;
  align-items: center;
  width: 300px;
  justify-content: flex-end;
  @media screen and (max-width: 767px) {
    width: 150px;
  }
}
.username {
  padding: 2px;
  border-radius: 3px;
  margin-right: 5px;
  text-decoration: none;
  color: $text-color;
  &:hover {
    color: white;
    background-color: $button-color-hover;
  }
  @include focus;
}
.clock {
  font-weight: 500;
}
