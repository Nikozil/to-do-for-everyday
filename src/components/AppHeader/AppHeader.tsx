import React, { MouseEvent } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router';
import { PageNames } from '../../constants/pageNames';
import { signOut } from '../../Redux/modules/userReducer';
import { AppStateType } from '../../Redux/store';
import styles from './AppHeader.module.scss';

const AppHeader = () => {
  const user = useSelector((state: AppStateType) => state.user.user);
  const pathname = useLocation().pathname;
  const dispatch = useDispatch();

  const signoutHandler = (event: MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    dispatch(signOut());
  };

  return (
    <header className={styles.header}>
      <div className={styles.pagename}>{PageNames[pathname]}</div>
      <div className={styles.usermenu}>
        <div className={styles.username}>
          {user ? user.displayName || user.email : 'Аноним'}
        </div>
        <button
          className={'btn btn-secondary btn-sm mx-1'}
          onClick={signoutHandler}>
          Выйти
        </button>
      </div>
    </header>
  );
};

export default AppHeader;
