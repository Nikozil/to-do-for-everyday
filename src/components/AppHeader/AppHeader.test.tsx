import React from 'react';
import { render, screen } from '../../utils/tests/test-utils';
import AppHeader from './AppHeader';

describe('AppHeader', () => {
  it('render correct', async () => {
    const { asFragment } = render(<AppHeader />, {
      preloadedState: {
        clock: { time: 1635263951643 },
      },
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('Clock render correct', () => {
    render(<AppHeader />, {
      preloadedState: {
        clock: { time: 1635263951643 },
      },
    });
    expect(screen.getByText(/18:59/i)).toBeInTheDocument();
  });
});
