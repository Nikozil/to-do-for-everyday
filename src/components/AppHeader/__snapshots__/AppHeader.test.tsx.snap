// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader render correct 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="pagename"
    />
    <div
      class="clock"
    >
       
    </div>
    <div
      class="usermenu"
    >
      <a
        class="username"
        href="/settings"
      >
        Аноним
      </a>
    </div>
  </header>
   
</DocumentFragment>
`;
